api/src/lang/parse.rs

```diff
@@ -1,101 +1,154 @@
 use crate::{
     error::CompileError,
     lang::{
-        ast::{Instr, LangValue, Program, StackIdentifier},
+        ast::{Instr, LangValue, Operator, Program, Register, StackIdentifier},
         Compiler,
     },
 };
 use nom::{
     branch::alt,
     bytes::complete::tag_no_case,
-    character::complete::{char, digit1, multispace0},
-    combinator::map_res,
-    multi::{many0, many1},
-    sequence::{delimited, preceded, terminated},
-    IResult,
+    character::complete::{char, digit1, multispace0, one_of},
+    combinator::{all_consuming, map_res},
+    multi::many0,
+    sequence::{delimited, preceded, tuple},
+    Compare, IResult, InputLength, InputTake,
 };
 
-fn parse_read(input: &str) -> IResult<&str, Instr> {
-    // tag_no_case returns a (str, str) tuple
-    // first element is whats left to parse
-    // second is what matched
-    let (input, _) = tag_no_case("Read")(input)?;
-    Ok((input, Instr::Read))
+fn arg_delim(input: &str) -> IResult<&str, char> {
+    one_of(" \t")(input)
```

Comment:
I dont think this will catch multiple spaces, if we want them to be able to do that

Reply:
Yeah so I wanted to make it match multiple spaces without matching newlines and just didn't bother figuring out the repeating part, I figured I'd leave that to you. Are you cool with leaving it for now and you can change it in your PR?

Reply:
yea thats fine